lib/spack/spack/test/concretize.py

```diff
@@ -2172,3 +2172,11 @@ def test_concretization_with_compilers_supporting_target_any(self):
         with spack.config.override("compilers", compiler_configuration):
             s = spack.spec.Spec("a").concretized()
         assert s.satisfies("%gcc@12.1.0")
+
+    def test_virtuals_are_annotated_on_edges(self, default_mock_concretization):
+        spec = default_mock_concretization("mpileaks ^mpich")
+
+        edges = spec.edges_to_dependencies(name="mpich")
+        assert len(edges) == 1 and edges[0].virtuals == ("mpi",)
+        edges = spec.edges_to_dependencies(name="callpath")
+        assert len(edges) == 1 and edges[0].virtuals == ()
```

Comment:
Can we make this test the case in which the dependency is found entirely by spack as well? Something like

```suggestion
    @pytest.mark.parametrize("spec", ["mpileaks", "mpileaks ^mpich"])
    def test_virtuals_are_annotated_on_edges(self, spec, default_mock_concretization):
        spec = default_mock_concretization(spec)

        mpiname = spec["mpi"].name
        edges = spec.edges_to_dependencies(name=mpiname)
        assert len(edges) == 1 and edges[0].virtuals == ("mpi",)
        edges = spec.edges_to_dependencies(name="callpath")
        assert len(edges) == 1 and edges[0].virtuals == ()
```

Reply:
Done in 29a8457ce561f956aa24067cb61bf080749f9050